src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Rules/AttributesMustMatch.cs

```diff
@@ -48,6 +48,32 @@ private static IEnumerable<string> ReadExclusions(IEnumerable<string> excludeAtt
             }
         }
 
+        private bool HasInternalArguments(AttributeData attr)
+        {
+            foreach (TypedConstant argument in attr.ConstructorArguments)
+            {
+                if (argument.Kind == TypedConstantKind.Type
+                    && argument.Value is INamedTypeSymbol typ
+                    && !typ.IsVisibleOutsideOfAssembly(_settings.IncludeInternalSymbols))
+                {
+                    return true;
+                }
+            }
+
+            foreach (KeyValuePair<string, TypedConstant> kv in attr.NamedArguments)
+            {
+                TypedConstant argument = kv.Value;
+                if (argument.Kind == TypedConstantKind.Type
+                    && argument.Value is INamedTypeSymbol typ
+                    && !typ.IsVisibleOutsideOfAssembly(_settings.IncludeInternalSymbols))
+                {
+                    return true;
+                }
+            }
```

Comment:
Both foreach loops are identical. Can you avoid the repetition (i.e. by using a local function)?